libdnf/rpm/solv_query.cpp

```diff
@@ -1794,9 +1794,13 @@ SolvQuery & SolvQuery::ifilter_installed() {
         Solvable * solvable = solv::get_solvable(pool, candidate_id);
         if (solvable->repo == installed_repo) {
             filter_result.add_unsafe(candidate_id);
+            continue;
+        }
+        if (candidate_id.id >= installed_repo->end) {
+            break;
         }
     }
-    p_impl->query_result &= filter_result;
+    p_impl->query_result = std::move(filter_result);
```

Comment:
I know that it is faster but in case that we will give const reference to `query_result` as API, we have to document it.

Reply:
To be honest, I'm not sure if we want this optimization. I prepared a comit to know your opinion.

Reply:
I added TODO.